src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type InputMap<T> = {
+  [key: string]: T;
+};
+
+export function groupByKey(
+  items: InputMap<string>[], key: string,
+): GroupsMap<object> {
```

Comment:
Твое решение рабочее, однако мы могли обойтись без создания дополнительного типа InputMap, который во многом повторяет тип GroupsMap.

```suggestion
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```
Здесь мы указываем, что наша функция будет работать с параметрами типа Т, аргумент items будет иметь тип массива с объектами типа Т, а аргумент key будет иметь тип ключа из объекта типа Т (это же и указано в условии задачи). А возвращать функция будет объект типа GroupsMap\<T>\.